### PR TITLE
[API Compatibility] support linspace, normal, nn.functional.dropout, nn.Dropout

### PR DESCRIPTION
**Pull request by LLSGYN**
### PR Category
User Experience

### PR Types
New features

### Description
+ paddle.normal: 支持out参数，size别名
+ paddle.linspace: 支持out，device，requires_grad参数，end, steps别名
+ dropout: 支持inplace

### CODE REVIEW

File: python/paddle/nn/functional/common.py [hunk 1]
```diff
@@ -1328,6 +1331,9 @@ def dropout(
                 seed if seed is not None else 0,
                 seed is not None,
             )
+            if inplace:
+                paddle.assign(out, x)
```
Comment (zhwesky2010) [inline python/paddle/nn/functional/common.py]: 这个没有走框架的inplace机制，会导致绕过错误检查，导致不可预知问题。

在ops.yaml中加一个inpace，这里使用`_C_ops.dropout_`会有什么问题

File: python/paddle/tensor/random.py [hunk 2]
```diff
@@ -1439,11 +1454,14 @@ def normal(
         mean = float(mean)
         out = standard_normal(paddle.shape(std), std.dtype, name)
     else:
-        return gaussian(shape=shape, mean=mean, std=std, name=name)
+        return gaussian(shape=shape, mean=mean, std=std, name=name, out=_out)
 
     out = out * std + mean
```
Comment (zhwesky2010) [inline python/paddle/tensor/random.py]: 上面的out是一个中间变量值，换一个名字吧。免得和传入的out混淆了，代码易读点

File: python/paddle/nn/functional/common.py [hunk 3]
```diff
@@ -1417,7 +1428,7 @@ def get_attrs(prog, dropout_prob, is_test, seed):
 
                 if isinstance(
                     dropout_prob, Variable
-                ) and not dropout_prob.shape != [1]:
+                ) and dropout_prob.shape != [1]:
```
Comment (zhwesky2010) [inline python/paddle/nn/functional/common.py:1431]: 这个改动原因是？

File: test/legacy_test/test_dropout_op_inplace.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                    training=False,
+                    inplace=True,
+                )
+                self.assertEqual(id(result), input_id)
+
+
+class TestDropoutInplacePIR(unittest.TestCase):
+    def test_pir_mode(self):
+        with (
+            paddle.pir_utils.IrGuard(),
```
Comment (zhwesky2010) [inline test/legacy_test/test_dropout_op_inplace.py:94]: 单测里你还得测一下**PIR**下的cinn模式，代码才能覆盖到 Dropout_OpInferSymbolicShape

**paddle.jit.to_static(cinn=True)，参考其他单测**

File: python/paddle/nn/functional/common.py (hunk 3 above)
Comment (LLSGYN) [reply python/paddle/nn/functional/common.py:1431]: 这里是我注意到的一个可能问题，这里的shape检查要求`shape == [1]`，但是之前的代码实现中的判断是`not dropout_prob.shape != [1]`，即`dropout_prob.shape == [1]`时会raise TypeError，和下面的描述相矛盾。
```python
raise TypeError(
    f"Required p.shape == [1] if type(p) is Variable, but received p.shape = {p.shape}"
)
```

**Review by zhwesky2010 — APPROVED** [review]
LGTM

**Review by SigureMo — APPROVED** [review]

**Review by zyfncg — APPROVED** [review]

**Review by jeff41404 — APPROVED** [review]
LGTM

**Comment by zhwesky2010** [comment]
@LLSGYN 这个CI / Distribute-stable-test 跑不过
